tools/pylib/boututils/showdata.py

```diff
@@ -396,8 +396,8 @@ def showdata(vars, titles=[], legendlabels = [], surf = [], polar = [], tslice =
 
 
         if not (global_colors):
-            if not float32(fmax[i])>float32(fmin[i]):
-                clevels.append([fmax[i],fmax[i]+1.])
+            if (fmax[i]-fmin[i])/abs(fmax[i]+fmin[i]) <= 2.e-15:
```

Comment:
Could this be done then with:

```python
from numpy import isclose
...
if isclose(fmin[i], fmax[i]):
    # Would be better to use a warning here
    print("Contour levels too close, using ", fmin[i])
    clevels.append(fmin[i])
else:
...
```
This will cause `contour` to plot only a single contour, but that's essentially what's being asked.

Reply:
Agreed isclose() is a nicer test :thumbsup:

clevels actually ends up being used by contourf(), not contour(). I was wrong when I wrote contour() above. contourf() needs at least two levels, which must be far enough apart. Keeping Ncolors contours as I have in the latest change allows us to see the difference between a totally constant field and one with O(1.e-15) variations. Maybe not much practical use, but it seems to work without throwing errors, so why not?

Reply:
It reads much better now, it's much clearer what's going on :+1: 

I'm not sure variations on the 1.e-15 level mean anything -- this is very close to the machine precision level -- but this looks fine as is.